plugins/test/framework.cc

```diff
@@ -134,22 +195,47 @@ TestHttpContext::Result TestHttpContext::SendRequestHeaders(
   phase_logs_.clear();
   result_ = Result{.headers = std::move(headers)};
   phase_ = proxy_wasm::WasmHeaderMapType::RequestHeaders;
-  result_.status =
+  current_callback_ = TestHttpContext::CallbackType::RequestHeaders;
+  result_.header_status =
       onRequestHeaders(result_.headers.size(), /*end_of_stream=*/false);
   phase_ = proxy_wasm::WasmHeaderMapType(-1);  // ideally 0 would mean unset
   return std::move(result_);
 }
+
+TestHttpContext::Result TestHttpContext::SendRequestBody(std::string body) {
+  phase_logs_.clear();
+  result_ = Result{};
+  body_buffer_.set(std::move(body));
+  current_callback_ = TestHttpContext::CallbackType::RequestBody;
+  result_.body_status =
+      onRequestBody(body_buffer_.size(), /*end_of_stream=*/false);
+  result_.body = body_buffer_.release();
+  return std::move(result_);
+}
+
 TestHttpContext::Result TestHttpContext::SendResponseHeaders(
     TestHttpContext::Headers headers) {
   phase_logs_.clear();
   result_ = Result{.headers = std::move(headers)};
   phase_ = proxy_wasm::WasmHeaderMapType::ResponseHeaders;
-  result_.status =
+  current_callback_ = TestHttpContext::CallbackType::ResponseHeaders;
+  result_.header_status =
       onResponseHeaders(result_.headers.size(), /*end_of_stream=*/false);
   phase_ = proxy_wasm::WasmHeaderMapType(-1);  // ideally 0 would mean unset
   return std::move(result_);
 }
 
+TestHttpContext::Result TestHttpContext::SendResponseBody(std::string body) {
+  phase_logs_.clear();
+  result_ = Result{};
+  body_buffer_.set(std::move(body));
```

Comment:
Same as earlier comment, this should call a new `SetOwned()` method.

Reply:
Same comment as earlier